include/measurement_kit/common/var.hpp

```diff
@@ -9,38 +9,61 @@
 
 namespace mk {
 
-template <typename T> class Var : public std::shared_ptr<T> {
-    using std::shared_ptr<T>::shared_ptr;
-
+template <typename T> class Var {
   public:
+    template <typename Deleter>
```

Comment:
Yes. The drawback is that we need to write more code. However, since Android build started to fail with the inheritance approach, I decided to move down this road.